packages/models/src/purpose/purpose.ts

```diff
@@ -96,4 +99,7 @@ export type RiskAnalysisDocumentPDFPayload = {
   freeOfChargeReason: string;
   date: string;
   eServiceMode: string;
+  producerDelegationId: string | undefined;
```

Comment:
To do the same we did for agreement contracts and delegation documents, I would:
- Move this from the commons `models` package to `purpose-process/src/model`, since it's used only there. It also simplifies development (you don't have to build models every time you test a change to the payload)
- replace `consumerText` and `producerText` with two variables each: name and code (for example, `consumerName` and `consumerCode`), so that the logic to display or not the code (IPA or another type in the future) is in the PDF template
- use branded types where it makes sense, to avoid error in swapping params (e.g., delegationId can be `DelegationId | undefined`

Reply:
I would also improve the tests to do what we did in the agreement / delegation processes: spy on `pdfGenerator.generate` to verify that the parameters passed to the template are correct: https://github.com/pagopa/interop-be-monorepo/blob/1b99bcfe93b3a5ac7b46a1081644447356158560/packages/agreement-process/test/activateAgreement.test.ts#L462